### ISSUE TITLE
Error:  Cannot instantiate cyclic dependency!

### ISSUE THREAD

**Issue by dxvladislavvolkov** [clbond/angular-ssr#29]
Hello!
I have the handler for the DOMContentLoaded event.
When I delete the event handler, I get the code recursion.

To reproduce the problem, you can use the following code.
```js
var callback = function() {
    console.log("callback");
}

var loadedCallback = function() {
    callback();
    document.removeEventListener("DOMContentLoaded", loadedCallback)
};
document.addEventListener("DOMContentLoaded", loadedCallback)
```
Error:
callback
callback
callback
callback
callback
callback
callback
callback
callback
[2017-05-02T19:32:00.196Z] [error] [] Failed to render application: Error:  caused by: Cannot instantiate cyclic dependency!: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.: Error during instantiation of DocumentContainer!.
    at wrappedError (e:\examples\111\angular-ssr-master\examples\cli\node_modules\@angular\core\bundles\core.umd.js:1193:34) [http://localhost/]
    at injectionError (e:\examples\111\angular-ssr-master\examples\cli\node_modules\@angular\core\bundles\core.umd.js:1239:52) [http://localhost/]
    at instantiationError (e:\examples\111\angular-ssr-master\examples\cli\node_modules\@angular\core\bundles\core.umd.js:1337:12) [http://localhost/]
    at ReflectiveInjector_._instantiate (e:\examples\111\angular-ssr-master\examples\cli\node_modules\@angular\core\bundles\core.umd.js:2725:19) [http://localhost/]
    at ReflectiveInjector_._instantiateProvider (e:\examples\111\angular-ssr-master\examples\cli\node_modules\@angular\core\bundles\core.umd.js:2698:25) [http://localhost/]
    at ReflectiveInjector_._new (e:\examples\111\angular-ssr-master\examples\cli\packages\core\src\di\reflective_injector.ts:349:4) [http://localhost/]
    at ReflectiveInjector_._getObj[...]

**Comment by clbond** [comment]
Can you point me to the code that this happens in?

**Comment by dxvladislavvolkov** [comment]
I prepared an example by simply adding these lines of code to app.module.ts
You can see the example [here](https://github.com/dxvladislavvolkov/angular-ssr-cli)

**Comment by clbond** [comment]
Thank you for your bug report. This issue is fixed in `0.1.104`.
